### PR TITLE
January ARM Release

### PR DESCRIPTION
**Pull request by marstr**
Pulling down ARM changes after updating to the latest bits in the AutoRest repository and re-generating ARM packages.

Once approved, Release notes will be added, and this will be shipped as a release on the master branch.

### CODE REVIEW

File: Gododir/gen.go [hunk 1]
```diff
@@ -49,6 +49,14 @@ var (
...
 					Version: "2016-05-16",
 				},
+				// {
+				// 	Name:    "apimanagement",
+				// 	Version: "2016-07-07",
+				// },
+				// {
+				// 	Name:    "apimdeployment",
+				// 	Version: "2016-07-07",
+				// },
```
Comment (mcardosos) [inline Gododir/gen.go]: Connot we gen those?

File: arm/analysisservices/servers.go [hunk 2]
```diff
@@ -68,33 +65,34 @@ func (client ServersOperationsClient) Create(resourceGroupName string, serverNam
 				{Target: "serverName", Name: validation.Pattern, Rule: `^[a-z][a-z0-9]*$`, Chain: nil}}},
 		{TargetValue: serverParameters,
 			Constraints: []validation.Constraint{{Target: "serverParameters.ServerProperties", Name: validation.Null, Rule: false,
-				Chain: []validation.Constraint{{Target: "serverParameters.ServerProperties.ProvisioningState", Name: validation.ReadOnly, Rule: true, Chain: nil},
+				Chain: []validation.Constraint{{Target: "serverParameters.ServerProperties.State", Name: validation.ReadOnly, Rule: true, Chain: nil},
```
Comment (mcardosos) [inline arm/analysisservices/servers.go]: Aren't we getting rid of this in this release?

File: arm/datalake-store/account/accountgroup.go [hunk 3]
```diff
@@ -25,19 +25,20 @@ import (
 	"net/http"
 )
 
-// Client is the creates an Azure Data Lake Store account management client.
-type Client struct {
+// GroupClient is the creates an Azure Data Lake Store account management
```
Comment (mcardosos) [inline arm/datalake-store/account/accountgroup.go]: This Client makes more sense to me as "Client" than as "GroupClient"

File: arm/iothub/models.go [hunk 4]
```diff
@@ -310,61 +298,60 @@ func (client EventHubConsumerGroupsListResult) EventHubConsumerGroupsListResultP
...
-	Status            JobStatus  `json:"status,omitempty"`
-	FailureReason     *string    `json:"failureReason,omitempty"`
-	StatusMessage     *string    `json:"statusMessage,omitempty"`
-	DeviceID          *string    `json:"deviceId,omitempty"`
-	ParentJobID       *string    `json:"parentJobId,omitempty"`
-}
-
-// JobResponseListResult is the list of JobResponse objects with a next link.
+	JobID             *string           `json:"jobId,omitempty"`
+	StartTimeUtc      *date.TimeRFC1123 `json:"startTimeUtc,omitempty"`
```
Comment (mcardosos) [inline arm/iothub/models.go]: Weird change. I wonder its cause

**Comment by mcardosos** [comment]
> // GroupClient is the provides operations for working with resources and

I think this is the only place where the change from Client to GroupClient makes sense

---
Refers to: arm/resources/resources/resourcesgroup.go:28 in 8615c72. [](commit_id = 8615c72ea9785701a56c193d71707d0db885078c, deletion_comment = False)

**Comment by marstr** [comment]
After chatting about it during a team meeting, we've agreed that there are a lot of breaking changes that do not provide enough merit in this PR. To fix this situation, we're working on combing through and finding the packages that have name changes, but no other value. We will regenerate again without including those packages and manually bump the version numbers to match the rest of the SDK.

**Comment by marstr** [comment]
@jhendrixMSFT @mcardosos I just finished regenerating, please take another look! :)

File: arm/analysisservices/version.go [hunk 5]
```diff
@@ -14,30 +14,47 @@ package analysisservices
...
-	minor = "0"
-	patch = "1"
-	// Always begin a "tag" with a dash (as per http://semver.org)
-	tag             = "-beta"
-	semVerFormat    = "%s.%s.%s%s"
-	userAgentFormat = "Azure-SDK-for-Go/%s arm-%s/%s"
+	major           = "8"
+	minor           = "0"
+	patch           = "0"
+	tag             = ""
```
Comment (mcardosos) [inline arm/analysisservices/version.go]: >"" [](start = 19, length = 2)

beta is missing

Comment (marstr) [inline arm/analysisservices/version.go]: Doh! Alright, I'll fix that now.

**Comment by mcardosos** [comment]
API management is missing :(

---
In reply to: [278124784](https://github.com/Azure/azure-sdk-for-go/pull/528#issuecomment-278124784) [](ancestors = 278124784)

File: Gododir/gen.go [hunk 6]
```diff
@@ -50,14 +50,22 @@ var (
...
 				},
 				{
-					Name:    "authorization",
-					Version: "2015-07-01",
-				},
-				{
-					Name:    "batch",
-					Version: "2015-12-01",
-					Swagger: "BatchManagement",
+					Name:    "apimanagement",
```
Comment (mcardosos) [inline Gododir/gen.go]: @amarzavery Do you know if API management will have a composite swagger?
@marstr If this service wont have a composite swagger, I would follow the same approach as we did with machinelearning and resources.

**Pull request merged**